sagemaker-featurestore/feature_store_kms_key_encryption.ipynb

```diff
@@ -0,0 +1,560 @@
+    "orders_data[\"EventTime\"] = pd.Series([current_time_sec]*len(orders_data), dtype=\"float64\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "customer_data.head()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "orders_data.head()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Load feature definitions to your feature group. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "customers_feature_group.load_feature_definitions(data_frame=customer_data)\n",
+    "orders_feature_group.load_feature_definitions(data_frame=orders_data)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## How to create an Online or Offline Feature Store that uses your KMS key for encryption?\n",
+    "\n",
+    "Below we create two feature groups, `customers_feature_group` and `orders_feature_group` respectively, and explain how use your KMS key to securely encrypt your data in your online or offline feature store. \n",
+    "\n",
+    "### How to create an Online Feature store with your KMS key? \n",
+    "To encrypt data in your online feature store, set `enable_online_store` to be `True` and specify your KMS key as parameter `online_store_kms_key_id`. You will need to substitute your Account number in `arn:aws:kms:us-east-1:123456789012:key/` replacing `123456789012` with your Account number.\n",
+    "\n",
+    "```\n",
+    "customers_feature_group.create(\n",
+    "    s3_uri=f\"s3://{s3_bucket_name}/{prefix}\",\n",
+    "    record_identifier_name=record_identifier_feature_name,\n",
+    "    event_time_feature_name=\"EventTime\",\n",
+    "    role_arn=role,\n",
+    "    enable_online_store=True, \n",
+    "    online_store_kms_key_id = 'arn:aws:kms:us-east-1:123456789012:key/'+ new_kms_key['KeyMetadata']['KeyId']\n",
+    ")\n",
+    "\n",
+    "orders_feature_group.create(\n",
+    "    s3_uri=f\"s3://{s3_bucket_name}/{prefix}\",\n",
+    "    record_identifier_name=record_identifier_feature_name,\n",
+    "    event_time_feature_name=\"EventTime\",\n",
+    "    role_arn=role,\n",
+    "    enable_online_store=True,\n",
+    "    online_store_kms_key_id = 'arn:aws:kms:us-east-1:123456789012:key/'+new_kms_key['KeyMetadata']['KeyId']\n",
+    ")\n",
+    "```\n",
+    "\n",
+    "### How to create an Offline Feature store with your KMS key? \n",
+    "Similar to the above, set `enable_online_store` to be `False` and then specify your KMS key as parameter `offline_store_kms_key_id`. You will need to substitute your Account number in `arn:aws:kms:us-east-1:123456789012:key/` replacing `123456789012` with your Account number.\n",
```

Comment:
""Set `enable_online_store` to `False` and then specify your KMS key as parameter `offline_store_kms_key_id`. You must substitute your account number in `arn:aws:kms:us-east-1:123456789012:key/`, replacing `123456789012` with your account number."